decidim-core/app/commands/decidim/amendable/reject.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Amendable
+    # A command with all the business logic to reject an amend.
+    class Reject < Rectify::Command
+      # Public: Initializes the command.
+      #
+      # form         - A form object with the params.
+      # amendable    - The resource that is being amended.
+      def initialize(form)
+        @form = form
+        @amendment = form.amendment
+        @amendable = form.amendable
+        @emendation = form.emendation
+      end
+
+      # Executes the command. Broadcasts these events:
+      #
+      # - :ok when everything is valid, together with the amend.
+      # - :invalid if the form wasn't valid and we couldn't proceed.
+      #
+      # Returns nothing.
+      def call
+        return broadcast(:invalid) if form.invalid?
+
+        transaction do
+          @emendation.update state: "rejected"
+          reject_amendment!
```

Comment:
Emendations have their own state which they inherit from the amendment table.

emendation.state == amendment.state
emendation.state != amendable.state (but can be)

But in all the commands, instead of just ignoring the emendation state, we update it, too.

Reply:
I guess we don't need to. I refactored the code. Thanks.